public/request/user/update.php

```diff
@@ -28,13 +29,36 @@
 if ($propertyType === UserAction::UpdatePermissions) {
     $response = SetAccountPermissionsJSON($foundSourceUser->User, $permissions, $targetUsername, $value);
 
-    // auto-apply forums permissions
-    if (
-        $response['Success']
-        && $value >= Permissions::JuniorDeveloper
-        && !$foundTargetUser->ManuallyVerified
-    ) {
-        setAccountForumPostAuth($foundSourceUser, $permissions, $foundTargetUser, authorize: true);
+    if ($response['Success']) {
+        // Auto-apply forums permissions.
+        if ($value >= Permissions::JuniorDeveloper && !$foundTargetUser->ManuallyVerified) {
+            setAccountForumPostAuth($foundSourceUser, $permissions, $foundTargetUser, authorize: true);
+        }
+
+        // Adjust attached roles.
+        if ($value <= Permissions::Unregistered) {
+            $foundTargetUser->roles()->detach();
+        }
+        if ($value === Permissions::Registered) {
```

Comment:
These should be elseifs or a switch statement. None of the cases overlap.

Reply:
Went with elseifs in latest, as the `<=` case is not as clean to handle with a switch.

Reply:
I was thinking the `<=` could be the default cause, but I can see why that wouldn't necessarily be desired.